src/app/index.run.coffee

```diff
@@ -100,3 +100,18 @@ angular.module 'mnoEnterpriseAngular'
         $rootScope.hasNoOrganisations = true
     )
   )
+
+# Prevent access to panes other than company and account when the currrent organization isn't active
+# See  https://github.com/angular-ui/ui-router/wiki/Frequently-Asked-Questions#how-to-create-rules-to-prevent-access-to-a-state
+  .run(($rootScope, $state, MnoeOrganizations) ->
+    $rootScope.$on('$stateChangeStart', (e, to) ->
+      MnoeOrganizations.getCurrentOrganisation().then( (org) ->
+        if !org.organization.active
+          return if to.name[0..3] != "home" || to.name == "home.company" || to.name == "home.account"
+          e.preventDefault()
```

Comment:
This solution works, but to be more consistent with the existing codebase, could you try to do something like this: https://github.com/maestrano/mno-enterprise-angular/blob/2.0/src/app/index.route.coffee#L41

Reply:
I am not convinced for the logic to be in `index.route` as it needs to be in each state you want to *forbid*  (namely `home.apps, home.impac, onboarding, home.marketplace, home.provisioning, home.subscriptions)`, whereas in `index.run` are simply listed the *reachable* routes. Thus any newly added new state will by default be forbidden.

 It works well for `loginRequired` because it englobes `home` and `onboarding` while the states we want to forbid here are a subset of `home`. Or  should the states be reworked to take the deletion process in mind?